### ISSUE TITLE
AnimeTorrents downloads HTML instead of Torrent files

### ISSUE THREAD

**Issue by DevYukine** [Prowlarr/Prowlarr#217]
**Describe the bug**

When downloading any Torrent from AnimeTorrents through Prowlarr it does actually download HTML instead the correct .torrent file

**To Reproduce**

1. Go to Search
2. Search for anything on AnimeTorrents
3. Download any Torrent file through Prowlarr
4. Check downloaded file, its actually HTML inside of the .torrent file instead the torrent data

**Expected behavior**

I expected to download the real .torrent file instead of HTML

**Screenshots**

Not needed

**Platform Information (please complete the following information):**
 - Docker:  Yes
 - .NET Version (System -> Status):  5.0.6
 - Browser and Version (Only needed for UI issues): 
 - Prowlarr Version:  0.1.0.458
 - Prowlarr Branch:  develop

Turn on Trace logs under Settings -> General and wait for the bug to occur again.
**Upload the full log file here (or another site (e.g. pastebin) and link it). Issues will be closed, if they do not include this!**

https://gist.github.com/DevYukine/adcd1b95e2686edb09423b6b85b71cbe

**Comment by Qstick** [comment]
OK so, this seems to be a problem with how we store cookies in `IndexerStatus` table. Cookies is a `IDictionary<string, string>` and our current `DictionaryKeyPolicy` for serialization is `DictionaryKeyPolicy = JsonNamingPolicy.CamelCase`. This results in the cookie key being converted to Camel Case which kills the cookie for some sites (it's expected to be exact case). 

This shows up in AnimeTorrents, the cookie keys get manipulated when we store them and the cookies don't work when we bring them back from db. This results in searches always having to auth, and downloads always failing because we don't check for auth and try to re-auth for them. 

So in short, we need to fix our serialization settings for the cookies dictionary or store them in another way.
